src/mapping/star_align/script.py

```diff
@@ -0,0 +1,161 @@
+import os
+import re
+import tempfile
+import subprocess
+from pathlib import Path
+import tarfile
+import gzip
+import shutil
+
+## VIASH START
+par = {
+  'genomeDir': 'resources_test/cellranger_tiny_fastq/cellranger_tiny_ref/star/',
+  'readFilesIn': [
+    'resources_test/cellranger_tiny_fastq/cellranger_tiny_fastq/tinygex_S1_L001_R1_001.fastq.gz',
+    'resources_test/cellranger_tiny_fastq/cellranger_tiny_fastq/tinygex_S1_L001_R2_001.fastq.gz',
+    'resources_test/cellranger_tiny_fastq/cellranger_tiny_fastq/tinygex_S1_L002_R1_001.fastq.gz',
+    'resources_test/cellranger_tiny_fastq/cellranger_tiny_fastq/tinygex_S1_L002_R2_001.fastq.gz'
+  ]
+}
+meta = {
+  'config': 'src/mapping/star_align/.config.vsh.yaml',
+  'cpus': None,
+  'memory_b': None,
+  'memory_kb': None,
+  'memory_mb': None,
+  'memory_gb': None,
+  'memory_tb': None,
+  'memory_pb': None,
+  'temp_dir': '/tmp'
+}
+## VIASH END
+
+# fastqgz_regex = r'([A-Za-z0-9\-_\.]+)_S(\d+)_L(\d+)_([RI]\d+)_(\d+)\.fastq(\.gz)?'
+fastqgz_regex = r'([A-Za-z0-9\-_\.]+)_S(\d+)_L(\d+)_(R\d+)_(\d+)\.fastq(\.gz)?'
+
+def is_gz_file(filepath):
```

Comment:
Could use https://docs.python.org/3/library/mimetypes.html for this

Reply:
This only looks at the path name, not at the file itself. I suppose that would also work, though it's not what I'm currently doing in the `is_gz_file(...)`.

Reply:
I linked the incorrect library. I used https://github.com/ahupp/python-magic in the past. But adding an extra dependency is not worth it.